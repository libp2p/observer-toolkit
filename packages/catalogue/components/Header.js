import React from 'react'
import T from 'prop-types'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import ReactMarkdown from 'react-markdown'

import { useHidePrevious, Icon, Tooltip } from '@nearform/observer-sdk'

const HeaderOuter = styled.div`
  display: flex;
  padding: ${({ theme }) => theme.spacing([0, 1])};
  height: ${({ theme }) => theme.spacing(8)};
  align-items: center;
  justify-content: space-between;
`

const LogoContainer = styled.div`
  display: flex;
  align-items: center;
  margin: ${({ theme }) => theme.spacing(1)};
  ${({ theme }) => theme.text('body', 'medium')};
  color: ${({ theme }) => theme.color('contrast', 1)};
  text-decoration: none;
`

const Logo = styled.img`
  height: 48px;
  display: block;
  margin-right: ${({ theme }) => theme.spacing(2)};
`

const NavTray = styled.nav`
  display: flex;
  align-items: center;
  margin: ${({ theme }) => theme.spacing(2)};
  ${({ theme }) => theme.text('body', 'large')};
  color: ${({ theme }) => theme.color('text', 2)};
  font-weight: 600;
`

const NavLink = styled.a`
  display: block;
  white-space: nowrap;
  margin: ${({ theme }) => theme.spacing([0, 2])};
  text-decoration: none;
  color: ${({ theme }) => theme.color('text', 2)};
`

const StyledLink = styled(Link)`
  color: ${({ theme }) => theme.color('contrast', 1)};
  text-decoration: none;
`

const ExpandIcon = styled.span`
  transform: rotate(90deg);
  [data-tooltip='open'] > * > & {
    transform: rotate(180deg);
  }
  ${({ theme }) => theme.transition({ property: 'transform' })}
  margin-bottom: 2px;
`

const ContentTooltip = styled.div`
  min-width: 320px;
  padding: ${({ theme }) => theme.spacing([2, 4])};
  ${({ theme }) => theme.text('body', 'medium')};
  font-weight: 400;
  & a {
    font-weight: 600;
    text-decoration: none;
  }
  & a:link {
    color: ${({ theme }) => theme.color('contrast', 1)};
  }
  & a:visited {
    color: ${({ theme }) => theme.color('contrast', 2)};
  }
`

function Header({ title, content = [] }) {
  const hidePrevious = useHidePrevious()
  return (
    <HeaderOuter>
      <StyledLink
        to={{
          pathname: '/',
          state: { fromLink: true },
        }}
      >
        <LogoContainer>
          <Logo src="logo_medium.png" /> {title}
        </LogoContainer>
      </StyledLink>
      <NavTray>
        {content.map(
          item =>
            (item.content && (
              <Tooltip
                content={<ReactMarkdown source={item.content} />}
                side="bottom"
                toleranceY={null}
                fixOn="no-hover"
                hidePrevious={hidePrevious}
                override={{ Content: ContentTooltip }}
              >
                <NavLink>
                  {item.title}
                  <Icon type="expand" override={{ Container: ExpandIcon }} />
                </NavLink>
              </Tooltip>
            )) ||
            (item.url && (
              <NavLink href={item.url} target="_blank">
                {item.title}
              </NavLink>
            )) ||
            ''
        )}
      </NavTray>
    </HeaderOuter>
  )
}

Header.propTypes = {
  title: T.string,
  content: T.array,
}

export default Header
