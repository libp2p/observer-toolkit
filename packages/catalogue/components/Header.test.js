import React from 'react'

import { renderWithTheme } from '@nearform/observer-testing'

import Header from './Header'

describe('Header', () => {
  it('renders as expected', () => {
    const { asFragment } = renderWithTheme(<Header />)
    expect(asFragment()).toMatchSnapshot()
  })
})
