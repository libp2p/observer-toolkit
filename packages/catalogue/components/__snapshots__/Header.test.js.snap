// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected 1`] = `
<DocumentFragment>
  .c0 {
  width: inherit;
  height: inherit;
  display: inherit;
  -webkit-flex-direction: inherit;
  -ms-flex-direction: inherit;
  flex-direction: inherit;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px 8px;
  height: 64px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px;
  font-family: plex-sans,sans-serif;
  line-height: 1.6em;
  font-size: 10pt;
  line-height: 1.6em;
  color: rgb(78,23,91);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  height: 48px;
  display: block;
  margin-right: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 16px;
  font-family: plex-sans,sans-serif;
  line-height: 1.6em;
  font-size: 11pt;
  line-height: 1.6em;
  color: rgb(128,130,133);
  font-weight: 600;
}

.c2 {
  color: rgb(78,23,91);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/"
      >
        <div
          class="c3"
        >
          <img
            class="c4"
            src="logo_medium.png"
          />
           
        </div>
      </a>
      <nav
        class="c5"
      />
    </div>
  </div>
</DocumentFragment>
`;
